### DISCUSSION TITLE
[question] mount moosefs case-insensitive

### DISCUSSION THREAD

**Discussion by eleaner** [moosefs/moosefs#385] [Q&A]
Hi,

If I understand correctly moosefs is by nature case-sensitive
Mounted on my mac it retains this quality. As a result Lightroom classic works with .xmp files.
The same moosefs mount is then exposed to windows virtual machine, where another software happily works with .XMP
as you can imagine, as a result I end up with two sets of files that were supposed to be one.
Any idea how to go around it?

Thanks
Marcin

**Comment by chogata** [comment]
MooseFS currently has no option to work case insensitive.

Would it help if the files on your mac got .XMP extension, so later the windows virtual machine saw them as such? If yes, this could probably be achieved with automated apple script hooked to the folder where the files are created.

**Comment by eleaner** [comment]
sadly Lightroom ignores .XMP
so far the only sort-of working solution is to access moosefs on windows
via SMB share
but I am obviously hit on the performance

**Comment by chogata** [comment]
Another idea then :) 

Mount this share also somewhere with a Linux, have a script that runs in crontab every minute (if this is enough, idk how soon you access the files on one OS after creating them on another), lists all files and if it finds 'somefile.xmp', it creates a hardlink (if not exists) with name 'somefile.XMP'. And vice versa. You still have two names for each file, but at least it will be one file, not two separate files.

We have marked it in our notes to think about possible option for mfsmount to mount MooseFS case insensitive. It is a complicated problem, because unlike with a system that is either always case sensitive or always case insensitive, a system that is this on that machine and the other on the second machine must resolve many conflicts and avoid many pitfalls. So we cannot offer an immediate solution.

**Comment by eleaner** [comment]
thank you
